test/unit/org/apache/cassandra/io/sstable/SSTableIdTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.sstable;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.Sets;
+import com.google.common.primitives.UnsignedBytes;
+import org.junit.Test;
+
+import org.apache.cassandra.utils.UUIDGen;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.quicktheories.QuickTheory.qt;
+import static org.quicktheories.generators.SourceDSL.longs;
+
+public class SSTableIdTest
+{
+    @Test
+    public void testSequenceBasedIdProperties()
+    {
+        testSSTableIdProperties(SequenceBasedSSTableId.Builder.instance);
+    }
+
+    @Test
+    public void testUUIDBasedIdProperties()
+    {
+        testSSTableIdProperties(UUIDBasedSSTableId.Builder.instance);
+    }
+
+    private void testSSTableIdProperties(SSTableId.Builder<?> builder)
```

Comment:
I understand that another property of `SSTableId.Builder` is that the ids produced by a generator are unique even with concurrent calls to the supplier returned by `SSTableId.Builder#generator`. Maybe we could test this by concurrently generating some ids with a generator, and then verifying that there aren't any repetitions? For example with something like:
```java
private static <T extends SSTableId> void test(SSTableId.Builder<T> builder) throws Throwable
{
    Set<SSTableId> ids = new CopyOnWriteArraySet<>();
    Supplier<T> generator = builder.generator(ids.stream());
    ExecutorService service = Executors.newFixedThreadPool(NUM_THREADS);
    CountDownLatch latch = new CountDownLatch(1);
    for (int i = 0; i < NUM_THREADS; i++)
    {
        service.submit(() -> {
            try
            {
                Assert.assertTrue(latch.await(1, TimeUnit.MINUTES));
            }
            catch (Throwable t)
            {
                throw new RuntimeException(t);
            }
            for (int j = 0; j < IDS_PER_THREAD; j++)
                ids.add(generator.get());
        });
    }
    latch.countDown();
    service.shutdown();
    Assert.assertTrue(service.awaitTermination(10, TimeUnit.MINUTES));
    Assert.assertEquals(NUM_THREADS * IDS_PER_THREAD, ids.size());
}
```